internal/endorser/endorser_test.go

```diff
@@ -92,13 +99,80 @@ func TestGenerateEndorsement_FailingSingleRemoteProvenanceEndorsement(t *testing
 	}
 
 	provenances := []string{"https://github.com/project-oak/transparent-release/blob/main/testdata/provenance.json"}
-	_, err := GenerateEndorsement(binaryHash, validity, provenances)
+	reference := verifier.ProvenanceIR{
+		BinarySHA256Digests: []string{binaryHash},
+	}
+	_, err := GenerateEndorsement(reference, validity, provenances)
 	want := "could not load provenances"
 	if err == nil || !strings.Contains(err.Error(), want) {
 		t.Fatalf("got %q, want error message containing %q,", err, want)
 	}
 }
 
+func TestGenerateEndorsement_MultipleNotVerifiedOrConsistent(t *testing.T) {
```

Comment:
Added tests. Could you have a look @rbehjati ?

Reply:
Thanks. This test checks several scenarios. Since the scenarios do not require heavy set up, I prefer separate test cases, one for each scenario. That way when a test fails it will be easier to figure out which scenario failed.